packages/generator/ni_measurement_plugin_sdk_generator/client/__init__.py

```diff
@@ -89,91 +301,48 @@ def create_client(
     built_in_import_modules: List[str] = []
     custom_import_modules: List[str] = []
 
-    if all:
-        measurement_service_class = get_all_registered_measurement_service_classes(discovery_client)
-        if len(measurement_service_class) == 0:
-            raise click.ClickException("No registered measurements.")
-    else:
-        if not measurement_service_class:
-            raise click.ClickException(
-                "The measurement service class cannot be empty. Either provide a measurement service class or use the 'all' flag to generate clients for all registered measurements."
-            )
-
-    if directory_out is None:
-        directory_out_path = pathlib.Path.cwd()
+    has_batch_parameters = (
+        measurement_service_class or all or module_name or class_name or directory_out
```

Comment:
It should only take `measurement_service_class` or `all` into account.

In cases where users provide only the module name, or class name, or directory name, the has_batch_parameters boolean will still be set. However, without any of the following three commands (service_class, all, or interactive), we can't proceed with client generation.